DFE-Digital/sts-plan-technology-for-your-school
# feat: Modify ButtonWithEntryReference slug retrieval

## Overview

Use a view to retrieve `ButtonWithEntryReferenceDbEntity`s, with the view automatically populating the `Slug` from the `LinkToEntry` relationship.

## Changes

### Minor

- Create SQL view to retrieve `ButtonWithEntryReferences`, along with the `Slug` of their `LinkToEntry` relationship, and a `LinkType` column to indicate what type the related entity is. (This is used in the view to know how to generate the link)
- Modify the `ButtonWithEntryReferenceDbEntity` and `ButtonWithEntryReference` classes accordingly
- Modify the `CmsDbContext` to use the view, and to ignore the new properties on saving

## How to review the PR

Run the app from this PR.

1. Ensure that `ButtonWithEntryReference` contents still work.
2. Ensure that the view is called; check debug output for something like `INNER JOIN [Contentful].[ButtonWithEntryReferencesWithSlug]`
3. Modify `ButtonWithEntryReference` in Contentful; ensure that the app processes the requests and updates the DB accordingly. Test not just field changes, but also publish/unpublish/etc
4. Create `ButtonWithEntryReference` in Contentful, ensure that the app creates it in the DB as expected.

## Additional notes

Actually don't think I like this as now we've got 2 places to update if we add a new content type to the `LinkToEntry` field but... I dunno.

## Checklist

Delete any rows that do not apply to the PR.
